crates/api/src/module.rs

```diff
@@ -461,8 +404,20 @@ impl Module {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn imports(&self) -> &[ImportType] {
-        &self.inner.imports
+    pub fn imports<'me>(&'me self) -> impl Iterator<Item = ImportType> + 'me {
+        let module = self.inner.compiled.module_ref();
+        module
+            .imports
+            .iter()
+            .map(move |(module_name, name, entity_index)| {
+                let r#type = entity_type(entity_index, module);
+                ImportType::new(module_name, name, r#type)
+            })
+    }
+
+    /// Return the number of imports in this module.
+    pub fn num_imports(&self) -> usize {
```

Comment:
I think if we return an `ExactSizeIterator` we can avoid adding an extra method for this, since it should still be possible to do `module.imports().len()` in that case.

Reply:
Cool, done.